decidim-admin/app/commands/decidim/admin/destroy_category.rb

```diff
@@ -30,7 +31,9 @@ def call
       attr_reader :category
 
       def destroy_category
-        category.destroy!
+        Decidim.traceability.perform_action!(:delete, category, @user) do
```

Comment:
This breaks deleting categories that are sub-categories.

If I try to delete a sub-category after applying these changes I get the following error:

<details>
  <summary>FrozenError (can't modify frozen attributes) [click to see the stacktrace]</summary>
  
```
FrozenError (can't modify frozen attributes):
  
activemodel (6.1.5) lib/active_model/attribute_set.rb:50:in `write_from_user'
activerecord (6.1.5) lib/active_record/attribute_methods/write.rb:34:in `write_attribute'
activerecord (6.1.5) lib/active_record/attribute_methods.rb:345:in `[]='
activerecord (6.1.5) lib/active_record/associations/belongs_to_association.rb:114:in `replace_keys'
activerecord (6.1.5) lib/active_record/associations/belongs_to_polymorphic_association.rb:18:in `replace_keys'
activerecord (6.1.5) lib/active_record/associations/belongs_to_association.rb:82:in `replace'
activerecord (6.1.5) lib/active_record/associations/singular_association.rb:17:in `writer'
activerecord (6.1.5) lib/active_record/associations/builder/association.rb:111:in `participatory_space='
/decidim/decidim-core/app/models/decidim/category.rb:63:in `subcategories_have_same_participatory_space'
activesupport (6.1.5) lib/active_support/callbacks.rb:427:in `block in make_lambda'
activesupport (6.1.5) lib/active_support/callbacks.rb:198:in `block (2 levels) in halting'
activesupport (6.1.5) lib/active_support/callbacks.rb:604:in `block (2 levels) in default_terminator'
activesupport (6.1.5) lib/active_support/callbacks.rb:603:in `catch'
activesupport (6.1.5) lib/active_support/callbacks.rb:603:in `block in default_terminator'
activesupport (6.1.5) lib/active_support/callbacks.rb:199:in `block in halting'
activesupport (6.1.5) lib/active_support/callbacks.rb:512:in `block in invoke_before'
activesupport (6.1.5) lib/active_support/callbacks.rb:512:in `each'
activesupport (6.1.5) lib/active_support/callbacks.rb:512:in `invoke_before'
activesupport (6.1.5) lib/active_support/callbacks.rb:105:in `run_callbacks'
activesupport (6.1.5) lib/active_support/callbacks.rb:824:in `_run_validation_callbacks'
activemodel (6.1.5) lib/active_model/validations/callbacks.rb:117:in `run_validations!'
activemodel (6.1.5) lib/active_model/validations.rb:337:in `valid?'
activerecord (6.1.5) lib/active_record/validations.rb:68:in `valid?'
/decidim/decidim-core/app/services/decidim/traceability.rb:120:in `log'
/decidim/decidim-core/app/services/decidim/traceability.rb:66:in `block (2 levels) in perform_action!'
activerecord (6.1.5) lib/active_record/connection_adapters/abstract/database_statements.rb:320:in `block in transaction'
activerecord (6.1.5) lib/active_record/connection_adapters/abstract/transaction.rb:319:in `block in within_new_transaction'
activesupport (6.1.5) lib/active_support/concurrency/load_interlock_aware_monitor.rb:26:in `block (2 levels) in synchronize'
activesupport (6.1.5) lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `handle_interrupt'
activesupport (6.1.5) lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `block in synchronize'
activesupport (6.1.5) lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `handle_interrupt'
activesupport (6.1.5) lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `synchronize'
activerecord (6.1.5) lib/active_record/connection_adapters/abstract/transaction.rb:317:in `within_new_transaction'
activerecord (6.1.5) lib/active_record/connection_adapters/abstract/database_statements.rb:320:in `transaction'
activerecord (6.1.5) lib/active_record/transactions.rb:209:in `transaction'
/decidim/decidim-core/app/services/decidim/traceability.rb:63:in `block in perform_action!'
paper_trail (12.3.0) lib/paper_trail/request.rb:107:in `with'
paper_trail (12.3.0) lib/paper_trail.rb:77:in `request'
/decidim/decidim-core/app/services/decidim/traceability.rb:62:in `perform_action!'
/decidim/decidim-admin/app/commands/decidim/admin/destroy_category.rb:34:in `destroy_category'
/decidim/decidim-admin/app/commands/decidim/admin/destroy_category.rb:25:in `call'
/decidim/decidim-core/lib/decidim/command.rb:18:in `call'
/decidim/decidim-admin/app/controllers/decidim/admin/categories_controller.rb:70:in `destroy'
actionpack (6.1.5) lib/action_controller/metal/basic_implicit_render.rb:6:in `send_action'
actionpack (6.1.5) lib/abstract_controller/base.rb:228:in `process_action'
actionpack (6.1.5) lib/action_controller/metal/rendering.rb:30:in `process_action'
actionpack (6.1.5) lib/abstract_controller/callbacks.rb:42:in `block in process_action'
activesupport (6.1.5) lib/active_support/callbacks.rb:117:in `block in run_callbacks'
activesupport (6.1.5) lib/active_support/core_ext/time/zones.rb:66:in `use_zone'
/decidim/decidim-core/app/controllers/concerns/decidim/use_organization_time_zone.rb:21:in `use_organization_time_zone'
activesupport (6.1.5) lib/active_support/callbacks.rb:126:in `block in run_callbacks'
i18n (1.10.0) lib/i18n.rb:328:in `with_locale'
/decidim/decidim-core/app/controllers/concerns/decidim/locale_switcher.rb:24:in `switch_locale'
activesupport (6.1.5) lib/active_support/callbacks.rb:126:in `block in run_callbacks'
activesupport (6.1.5) lib/active_support/callbacks.rb:137:in `run_callbacks'
actionpack (6.1.5) lib/abstract_controller/callbacks.rb:41:in `process_action'
actionpack (6.1.5) lib/action_controller/metal/rescue.rb:22:in `process_action'
actionpack (6.1.5) lib/action_controller/metal/instrumentation.rb:34:in `block in process_action'
activesupport (6.1.5) lib/active_support/notifications.rb:203:in `block in instrument'
activesupport (6.1.5) lib/active_support/notifications/instrumenter.rb:24:in `instrument'
activesupport (6.1.5) lib/active_support/notifications.rb:203:in `instrument'
actionpack (6.1.5) lib/action_controller/metal/instrumentation.rb:33:in `process_action'
actionpack (6.1.5) lib/action_controller/metal/params_wrapper.rb:249:in `process_action'
activerecord (6.1.5) lib/active_record/railties/controller_runtime.rb:27:in `process_action'
actionpack (6.1.5) lib/abstract_controller/base.rb:165:in `process'
actionview (6.1.5) lib/action_view/rendering.rb:39:in `process'
actionpack (6.1.5) lib/action_controller/metal.rb:190:in `dispatch'
actionpack (6.1.5) lib/action_controller/metal.rb:254:in `dispatch'
actionpack (6.1.5) lib/action_dispatch/routing/route_set.rb:50:in `dispatch'
actionpack (6.1.5) lib/action_dispatch/routing/route_set.rb:33:in `serve'
actionpack (6.1.5) lib/action_dispatch/journey/router.rb:50:in `block in serve'
actionpack (6.1.5) lib/action_dispatch/journey/router.rb:32:in `each'
actionpack (6.1.5) lib/action_dispatch/journey/router.rb:32:in `serve'
actionpack (6.1.5) lib/action_dispatch/routing/route_set.rb:842:in `call'
railties (6.1.5) lib/rails/engine.rb:539:in `call'
railties (6.1.5) lib/rails/railtie.rb:207:in `public_send'
railties (6.1.5) lib/rails/railtie.rb:207:in `method_missing'
actionpack (6.1.5) lib/action_dispatch/routing/mapper.rb:20:in `block in <class:Constraints>'
actionpack (6.1.5) lib/action_dispatch/routing/mapper.rb:49:in `serve'
actionpack (6.1.5) lib/action_dispatch/journey/router.rb:50:in `block in serve'
actionpack (6.1.5) lib/action_dispatch/journey/router.rb:32:in `each'
actionpack (6.1.5) lib/action_dispatch/journey/router.rb:32:in `serve'
actionpack (6.1.5) lib/action_dispatch/routing/route_set.rb:842:in `call'
railties (6.1.5) lib/rails/engine.rb:539:in `call'
railties (6.1.5) lib/rails/railtie.rb:207:in `public_send'
railties (6.1.5) lib/rails/railtie.rb:207:in `method_missing'
actionpack (6.1.5) lib/action_dispatch/routing/mapper.rb:20:in `block in <class:Constraints>'
actionpack (6.1.5) lib/action_dispatch/routing/mapper.rb:49:in `serve'
actionpack (6.1.5) lib/action_dispatch/journey/router.rb:50:in `block in serve'
actionpack (6.1.5) lib/action_dispatch/journey/router.rb:32:in `each'
actionpack (6.1.5) lib/action_dispatch/journey/router.rb:32:in `serve'
actionpack (6.1.5) lib/action_dispatch/routing/route_set.rb:842:in `call'
railties (6.1.5) lib/rails/engine.rb:539:in `call'
railties (6.1.5) lib/rails/railtie.rb:207:in `public_send'
railties (6.1.5) lib/rails/railtie.rb:207:in `method_missing'
actionpack (6.1.5) lib/action_dispatch/routing/mapper.rb:20:in `block in <class:Constraints>'
actionpack (6.1.5) lib/action_dispatch/routing/mapper.rb:49:in `serve'
actionpack (6.1.5) lib/action_dispatch/journey/router.rb:50:in `block in serve'
actionpack (6.1.5) lib/action_dispatch/journey/router.rb:32:in `each'
actionpack (6.1.5) lib/action_dispatch/journey/router.rb:32:in `serve'
actionpack (6.1.5) lib/action_dispatch/routing/route_set.rb:842:in `call'
batch-loader (1.5.0) lib/batch_loader/middleware.rb:11:in `call'
omniauth (2.1.0) lib/omniauth/strategy.rb:202:in `call!'
omniauth (2.1.0) lib/omniauth/strategy.rb:169:in `call'
omniauth (2.1.0) lib/omniauth/builder.rb:44:in `call'
bullet (7.0.1) lib/bullet/rack.rb:15:in `call'
rack-attack (6.6.1) lib/rack/attack.rb:127:in `call'
warden (1.2.9) lib/warden/manager.rb:36:in `block in call'
warden (1.2.9) lib/warden/manager.rb:34:in `catch'
warden (1.2.9) lib/warden/manager.rb:34:in `call'
/decidim/decidim-core/lib/decidim/middleware/strip_x_forwarded_host.rb:12:in `call'
/decidim/decidim-core/lib/decidim/middleware/current_organization.rb:22:in `call'
rack (2.2.3) lib/rack/tempfile_reaper.rb:15:in `call'
rack (2.2.3) lib/rack/etag.rb:27:in `call'
rack (2.2.3) lib/rack/conditional_get.rb:40:in `call'
rack (2.2.3) lib/rack/head.rb:12:in `call'
actionpack (6.1.5) lib/action_dispatch/http/permissions_policy.rb:22:in `call'
actionpack (6.1.5) lib/action_dispatch/http/content_security_policy.rb:19:in `call'
rack (2.2.3) lib/rack/session/abstract/id.rb:266:in `context'
rack (2.2.3) lib/rack/session/abstract/id.rb:260:in `call'
actionpack (6.1.5) lib/action_dispatch/middleware/cookies.rb:689:in `call'
activerecord (6.1.5) lib/active_record/migration.rb:601:in `call'
actionpack (6.1.5) lib/action_dispatch/middleware/callbacks.rb:27:in `block in call'
activesupport (6.1.5) lib/active_support/callbacks.rb:98:in `run_callbacks'
actionpack (6.1.5) lib/action_dispatch/middleware/callbacks.rb:26:in `call'
actionpack (6.1.5) lib/action_dispatch/middleware/executor.rb:14:in `call'
actionpack (6.1.5) lib/action_dispatch/middleware/actionable_exceptions.rb:18:in `call'
actionpack (6.1.5) lib/action_dispatch/middleware/debug_exceptions.rb:29:in `call'
web-console (4.2.0) lib/web_console/middleware.rb:132:in `call_app'
web-console (4.2.0) lib/web_console/middleware.rb:28:in `block in call'
web-console (4.2.0) lib/web_console/middleware.rb:17:in `catch'
web-console (4.2.0) lib/web_console/middleware.rb:17:in `call'
actionpack (6.1.5) lib/action_dispatch/middleware/show_exceptions.rb:33:in `call'
railties (6.1.5) lib/rails/rack/logger.rb:37:in `call_app'
railties (6.1.5) lib/rails/rack/logger.rb:26:in `block in call'
activesupport (6.1.5) lib/active_support/tagged_logging.rb:99:in `block in tagged'
activesupport (6.1.5) lib/active_support/tagged_logging.rb:37:in `tagged'
activesupport (6.1.5) lib/active_support/tagged_logging.rb:99:in `tagged'
railties (6.1.5) lib/rails/rack/logger.rb:26:in `call'
actionpack (6.1.5) lib/action_dispatch/middleware/remote_ip.rb:81:in `call'
request_store (1.5.1) lib/request_store/middleware.rb:19:in `call'
actionpack (6.1.5) lib/action_dispatch/middleware/request_id.rb:26:in `call'
rack (2.2.3) lib/rack/method_override.rb:24:in `call'
activesupport (6.1.5) lib/active_support/cache/strategy/local_cache_middleware.rb:29:in `call'
actionpack (6.1.5) lib/action_dispatch/middleware/executor.rb:14:in `call'
actionpack (6.1.5) lib/action_dispatch/middleware/static.rb:24:in `call'
actionpack (6.1.5) lib/action_dispatch/middleware/static.rb:24:in `call'
rack (2.2.3) lib/rack/sendfile.rb:110:in `call'
actionpack (6.1.5) lib/action_dispatch/middleware/host_authorization.rb:148:in `call'
rack-mini-profiler (3.0.0) lib/mini_profiler/profiler.rb:393:in `call'
webpacker (6.0.0.rc.5) lib/webpacker/dev_server_proxy.rb:25:in `perform_request'
rack-proxy (0.7.2) lib/rack/proxy.rb:67:in `call'
rack-cors (1.1.1) lib/rack/cors.rb:100:in `call'
railties (6.1.5) lib/rails/engine.rb:539:in `call'
puma (5.6.4) lib/puma/configuration.rb:252:in `call'
puma (5.6.4) lib/puma/request.rb:77:in `block in handle_request'
puma (5.6.4) lib/puma/thread_pool.rb:340:in `with_force_shutdown'
puma (5.6.4) lib/puma/request.rb:76:in `handle_request'
puma (5.6.4) lib/puma/server.rb:441:in `process_client'
puma (5.6.4) lib/puma/thread_pool.rb:147:in `block in spawn_thread'
```
</details>

I cannot replicate the same error at the `develop` branch.

Reply:
Nice catch @ahukkanen! We should add a spec for this case too then.

Reply:
I've fixed it in our case, but i'm not sure i see the big picture in this so i'd like your advice @ahukkanen